### PR TITLE
feat: implement user feedback loops for person grouping and model personalization

### LINKED ISSUE #188: roadmap: add user feedback loops for person grouping and model personalization
## Problem
Now that the People page exists, Find needs a careful feedback system where users can correct person clusters and eventually improve local model behavior.

## Why it matters
Different users have different libraries and expectations; the app should learn from local feedback without sending data away.

## Acceptance criteria
- [ ] Track person-cluster correction, general feedback capture, and low-compute personalization research as children.
- [ ] Keep all feedback local by default.
- [ ] Avoid promising heavy fine-tuning until research proves it is practical.
- [ ] Prioritize measurable improvements before model changes.

## Out of scope
Do not build cloud training, public datasets, or account-based personalization here.

## Suggested files
people page, face/person backend, feedback storage, ML pipeline docs.

## Notes for contributors
This is a parent roadmap. Start with feedback collection before model training.

### PR DESCRIPTION
**Pull request by KunjShah95**
## Overview
This PR adds the first foundation for local user feedback loops around People grouping, media feedback, and future personalization research.

## Scope
- Adds local feedback database models for person corrections and general ratings.
- Adds FastAPI endpoints for person split, merge, wrong-person, correct-group feedback, and search/caption/object ratings.
- Adds People page actions for split, merge, wrong-person, and correct-group feedback.
- Adds lightweight rating controls for search result relevance, caption quality, and object detection quality.
- Adds backend tests for feedback creation, validation, and stats.
- Adds research notes for low-compute personalization paths.

## Important GSSoC note
This PR closes the parent roadmap issue only. The child issues below are assigned to other contributors and should stay available for their own PRs unless those PRs fully complete the requested scope.

Closes #188.

Related / partially addressed foundation work:
- #189 - person cluster correction feedback
- #190 - manual feedback actions for search, captions, objects, and people
- #191 - low-compute personalization research

## Testing
- `uv run ruff check src/find_api/models/feedback.py src/find_api/routers/feedback.py src/find_api/routers/people.py tests/test_feedback.py`
- `uv run pytest tests/test_feedback.py -q`
- `pnpm exec biome check src/components/feedback-rating.tsx src/components/image-preview-modal.tsx src/app/search/page.tsx src/components/person-feedback-actions.tsx src/lib/api.ts src/app/people/page.tsx`
- `pnpm build`

## Summary by CodeRabbit

* **New Features**
  * Feedback system for person grouping corrections: split groups, merge with others, mark as wrong person, or confirm correct groupings
  * 1–5 star rating system for search results, captions, and object detection
  * Inline editing tools for caption and object label corrections to improve training data
  * Feedback management UI in person details with action buttons
  * Feedback analytics to track submitted corrections and average ratings

[Review Change Stack](https://app.coderabbit.ai/change-stack/Abhash-Chakraborty/Find/pull/197?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by Abhash-Chakraborty** [comment]
Fix the CI

**Comment by KunjShah95** [comment]
it has been resolved and checked you can now easily merge them  and assign appropriate labels

**Comment by KunjShah95** [comment]
@Abhash-Chakraborty  do check it

**Comment by KunjShah95** [comment]
KINDLY DO CHECK AND TELL ME WHATS THE UPDATE FOR THE ISSUES

**Comment by KunjShah95** [comment]
@Abhash-Chakraborty

**Comment by KunjShah95** [comment]
kindly do check these too

**Comment by KunjShah95** [comment]
@Abhash-Chakraborty  bhai is there anything more do I need to work on

**Comment by Abhash-Chakraborty** [comment]
I will review this now

**Review by Abhash-Chakraborty — APPROVED** [review]
Looks good now. The feedback flow is focused, the UI issues are cleaned up, and checks are passing.
